test/unit/io/sequence_file/sequence_file_output_test.cpp

```diff
@@ -372,6 +379,27 @@ TEST(row, different_fields_in_record_and_file)
     EXPECT_EQ(reinterpret_cast<std::ostringstream&>(fout.get_stream()).str(), expected_out);
 }
 
+TEST(row, writing_seq_qual)
+{
+    sequence_file_output fout{std::ostringstream{}, sequence_file_format_fasta{}, fields<field::ID, field::SEQ_QUAL>()};
+    fout.options.fasta_letters_per_line = 0;
+
+    std::vector<std::vector<qualified<dna5, phred42>>> seq_quals{3};
+    for (size_t i = 0; i < 3; ++i)
+    {
+        seq_quals[i].resize(quals[i].size());
+        for (size_t j = 0; j < quals[i].size(); ++j)
+        {
+            seq_quals[i][j] = seqs[i][j];
+            seq_quals[i][j] = quals[i][j];
+        }
```

Comment:
std::copy?

Also you don't need the vector of seq_quals if you immediatly emplace_back the current value anyway, right?